torch/testing/_internal/common_nn.py

```diff
@@ -5043,8 +5043,12 @@ def _do_test(self, test_case, module, input):
             # could probably unify check_jacobian above with this.
             params = tuple(x for x in module.parameters())
             num_inputs = len(input_tuple)
+            # These modules don't actually take the arguments as inputs so forward check will fail for these
+            # We will be able to re-enable this when https://github.com/pytorch/pytorch/issues/49171 is done
+            check_forward = len(params) == 0
             _assertGradAndGradgradChecks(
-                test_case, lambda *args, **kw: test_case._forward(module, args[:num_inputs]), input_tuple + params)
+                test_case, lambda *args, **kw: test_case._forward(module, args[:num_inputs]), input_tuple + params,
+                check_forward=check_forward)
```

Comment:
(no action required) Now, the very interesting thing here is that the grad and gradgrad checks only run if `check_gradgrad` is True! If `check_gradgrad` is False, then we don't run gradcheck, instead we go through `test_case.check_jacobian` above, which doesn't check forward-mode AD. `check_gradgrad=False` appears in a number of tests in test_nn, so we don't have coverage for those, but a lot of them weren't going to get the forward mode ad tests anyways because they're modules